notebooks/Notebook2.md

```diff
@@ -2,33 +2,115 @@
+
+```
+<dark>(treasure is north, cave is south, win is west) {
+    You find yourself in a dark room. A door leads to the north, while a cave opens to the south. A mysterious locked door also lies to the west. There's a beggar in the corner with a bucket of apples (maybe ask them about it?)
+    <beggar>(character) {
+        apple: Yes, those are my apples.
+        apples: Yes, those are my apples.
+        bucket: Do you like my bucket?
+        door: Go through that door if you dare
+    }
+}
+
+<cave>(dark is north) {
+    You've arrived in a dingy cave. You can go north to return to the starting room.
+}
+
+<treasure>() {
+    Wow, there's a treasure chest in here! You can't open it though (at least not yet). The starting room resides to the south.
+}
+
+<win>() {
+    Congratulations! You have escaped the dungeon and won the game.
+}
+```
+
+I then had to modify my parsing code to accept characters in the form written above. Once that was done, I had to build a lot of dictionaries on the backend. Each character needs a dialogue dictionary that stores things as key and corresponding text responses as value, while each room needs a dictionary that maps character names (ex. beggar or knight) to corresponding Character objects. It took a while, but that functionality is now complete. In a game, you can now use 'ask character about thing' syntax and it will print out the encoded responses if the character and thing both hit in the relevant dictionaries. If not, placeholder "I don't know who you're talking to"/"I don't know what you're talking about" text is printed instead.
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+The most pressing issues with my project are making decisions on the last few DSL things with items and then implementing it all in my artifact. This will be the bulk of the remaining work (aside from user testing and potential syntax revisions in the last week), so I'm expecting this to take the better part of two weeks to accomplish.
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+* Any feedback on my expanded DSL syntax and advice for the open questions on items would be very helpful! In particular, is the current verbose style of my syntax good for my audience (first-time programmers), and would a secondary, more succinct syntax be worth implementing?
+* Like last week, any comments on my parsing strategy would be nice. As I move into items, I probably will need to expand into pyparsing's parse actions functionality, so any words on that would be really appreciated.
```

Comment:
Your parsing strategy seems pretty good, as long as it works. It seems somewhat difficult to see how you would implement detailed error handling in it, however, which may be a concern.